libkmod/libkmod-module.c

```diff
@@ -672,16 +672,20 @@ static int do_init_module(struct kmod_module *mod, unsigned int flags, const cha
 
 		if (flags & KMOD_INSERT_FORCE_MODVERSION) {
 			err = kmod_elf_strip_section(elf, "__versions");
-			if (err < 0)
-				INFO(mod->ctx, "Failed to strip modversion: %s\n",
-				     strerror(-err));
+			if (err < 0 && err != -ENODATA) {
+				ERR(mod->ctx, "Failed to strip modversion: %s\n",
+				    strerror(-err));
+				return err;
+			}
 		}
 
 		if (flags & KMOD_INSERT_FORCE_VERMAGIC) {
 			err = kmod_elf_strip_vermagic(elf);
-			if (err < 0)
-				INFO(mod->ctx, "Failed to strip vermagic: %s\n",
-				     strerror(-err));
+			if (err < 0 && err != -ENODATA) {
```

Comment:
The .modinfo is optional: https://github.com/torvalds/linux/blob/master/kernel/module/main.c#L1829

Reply:
Discussing it here goes a bit side tracked, but:

1. Remove everything from mod-simple which needs external symbols:
```
diff --git a/testsuite/module-playground/mod-simple.c b/testsuite/module-playground/mod-simple.c
index 503e4d8..04d692f 100644
--- a/testsuite/module-playground/mod-simple.c
+++ b/testsuite/module-playground/mod-simple.c
@@ -1,28 +1,13 @@
-#include <linux/debugfs.h>
 #include <linux/init.h>
 #include <linux/module.h>

-static struct dentry *debugfs_dir;
-
-static int test_show(struct seq_file *s, void *data)
-{
-       seq_puts(s, "test");
-       return 0;
-}
-
-DEFINE_SHOW_ATTRIBUTE(test);
-
 static int __init test_module_init(void)
 {
-       debugfs_dir = debugfs_create_dir(KBUILD_MODNAME, NULL);
-       debugfs_create_file("test", 0444, debugfs_dir, NULL, &test_fops);
-
        return 0;
 }

 static void test_module_exit(void)
 {
-       debugfs_remove_recursive(debugfs_dir);
 }

 module_init(test_module_init);
```
2. Compile test suite
```
make check
```
3. Pick one of the mod-simple.ko files
```
cp $(find -name mod-simple.ko | head -n 1) .
```
3. Remove .modinfo
```
sed -i s/modinfo/mmminfo/ mod-simple.ko
```
5. Insert this module
```
sudo insmod -f mod-simple.ko
```
6. Verify that it's loaded
```
$ lsmod
Module                  Size  Used by
mod_simple             12288  0
```

Reply:
Granted, it doesn't look like it's really intended. But that's definitely out of scope for kmod, but a LKML discussion. Current behavior should still be possible, no matter how flawed such a setup really is. "It works."

Reply:
Isn't it too much chatter for a single if-condition, especially since this verification code depends on the kernel and it's their task to finally decide if a module shall be loaded or not? We've done our task: version information is gone. If the module was crap to begin with is nothing we decide about.